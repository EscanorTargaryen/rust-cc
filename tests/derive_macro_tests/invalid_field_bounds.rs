use rust_cc::*;

struct DoesNotImplementTrace;

#[derive(Trace, Finalize)]
struct MyStruct1 {
    field: DoesNotImplementTrace,
}

#[derive(Trace, Finalize)]
enum MyEnum3 {
    A(DoesNotImplementTrace),
}

fn main() {
}
