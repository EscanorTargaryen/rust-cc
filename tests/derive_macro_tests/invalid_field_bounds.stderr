error[E0277]: the trait bound `DoesNotImplementTrace: _::rust_cc::Trace` is not satisfied
 --> tests/derive_macro_tests/invalid_field_bounds.rs:5:10
  |
5 | #[derive(Trace, Finalize)]
  |          ^^^^^ the trait `_::rust_cc::Trace` is not implemented for `DoesNotImplementTrace`
  |
  = help: the following other types implement trait `_::rust_cc::Trace`:
            bool
            char
            isize
            i8
            i16
            i32
            i64
            i128
          and $N others
  = note: this error originates in the derive macro `Trace` (in Nightly builds, run with -Z macro-backtrace for more info)

error[E0277]: the trait bound `DoesNotImplementTrace: _::rust_cc::Trace` is not satisfied
  --> tests/derive_macro_tests/invalid_field_bounds.rs:10:10
   |
10 | #[derive(Trace, Finalize)]
   |          ^^^^^ the trait `_::rust_cc::Trace` is not implemented for `DoesNotImplementTrace`
   |
   = help: the following other types implement trait `_::rust_cc::Trace`:
             bool
             char
             isize
             i8
             i16
             i32
             i64
             i128
           and $N others
   = note: this error originates in the derive macro `Trace` (in Nightly builds, run with -Z macro-backtrace for more info)
